### PR TITLE
Don’t compile when Opcache is not enabled on CLI

### LINKED ISSUE #20878: [Cache] Zend OPcache seems to be disabled, can't compile file
I'm running Symfony 3.2 on a PHP 7.1 server with Opcache enabled for FPM but not for CLI.

Every time we deploy, this errors gets logged to Sentry:

```
Symfony\Component\Debug\Exception\FatalErrorException: Zend OPcache seems to be disabled, can't compile file
#1 /vendor/symfony/symfony/src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php(118): null
#0 Hautelook\SentryClient\ErrorHandler(null): Hautelook\SentryClient\ErrorHandler::handleFatalError
```

I think this is due the fact that I have `opcache.enable_cli => Off`. Since the warmup command runs on CLI, it will never work.

Maybe it would be good if the PHPFilesAdapter checks this configuration first, before trying to save it?

### PR DESCRIPTION
**Pull request by ruudk**
| Q             | A
| ------------- | ---
| Branch?       | 3.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #20878
| License       | MIT

This should fix #20878 "Zend OPcache seems to be disabled, can't compile file" when Opcache is enabled, but `opcache.enable_cli` is turned off.

### CODE REVIEW

File: src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php [hunk 1]
```diff
@@ -109,15 +109,28 @@ protected function doSave(array $values, $lifetime)
                     $value = serialize($value);
                 }
             } elseif (!is_scalar($value)) {
-                throw new InvalidArgumentException(sprintf('Value of type "%s" is not serializable', $key, gettype($value)));
+                throw new InvalidArgumentException(sprintf('Value of type "%s" is not serializable', gettype($value)));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php]: Borrowed from PhpArrayAdapter:
throw new InvalidArgumentException(sprintf('Cache key "%s" has non-serializable %s value.', $key, gettype($value)));

File: src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php [hunk 2]
```diff
@@ -109,15 +109,28 @@ protected function doSave(array $values, $lifetime)
...
 
         return $ok;
     }
+
+    /**
+     * @param string $file
+     */
+    private function compileFile($file)
+    {
+        if ('cli' === php_sapi_name() && !ini_get('opcache.enable_cli')) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php]: Use   PHP_SAPI constant instead

File: src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php [hunk 3]
```diff
@@ -109,15 +109,28 @@ protected function doSave(array $values, $lifetime)
...
+            $this->compileFile($file);
         }
 
         return $ok;
     }
+
+    /**
+     * @param string $file
+     */
+    private function compileFile($file)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php]: Not sure the extra method is worth it.

Comment (ruudk) [inline src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php]: Ok, removed it.

**Comment by fabpot** [comment]
Thank you @ruudk.

**Comment by nicolas-grekas** [comment]
Culprit: https://github.com/hautelook/sentry-client/issues/20

**Pull request closed without merging**